pkg/service/running_backups_registry.go

```
@@ -208,16 +209,35 @@ func (r *RunningBackupsRegistryImpl) GetRoutineState(routineName string) *model.
 	lastRun, found := r.lastSuccessful.Load(routineName)
 	if !found {
 		slog.Info("No last backup info available", slog.String("routine", routineName))
-		lastRun = &model.LastBackupRun{}
+		lastRun = model.NewNoBackupTime()
 	}
 
 	return &model.RoutineState{
 		Full:        currentBackupStatus(fullBackupHandler),
 		Incremental: currentBackupStatus(incrBackupHandler),
 		LastRunTime: lastRun,
+		NextRunTime: nextBackup(routineName, r.config),
 	}
 }
 
+func nextBackup(routineName string, config *model.Config) *model.BackupTime {
+	// at this moment routine exists and is validated.
+	routine, _ := config.Routine(routineName)
+	nextFullBackup := nextTrigger(routine.IntervalCron)
+	if routine.IncrIntervalCron == "" {
+		return model.NewFullBackupTime(nextFullBackup)
+	}
+
+	nextIncrementalBackup := nextTrigger(routine.IncrIntervalCron)
+	return model.NewBackupTime(nextFullBackup, nextIncrementalBackup)
+}
+
+func nextTrigger(cron string) time.Time {
+	trigger, _ := quartz.NewCronTrigger(cron)
+	fireTime, _ := trigger.NextFireTime(time.Now().UnixNano())
```

Do not ignore the errors.